code/modules/modular_computers/computers/machinery/console_presets.dm

```diff
@@ -19,96 +20,77 @@
+	install_programs(get_preset_programs(_app_preset_name))
 
-// Override in child types to install preset-specific programs.
-/obj/machinery/modular_computer/console/preset/proc/install_programs()
-	return
+/obj/machinery/modular_computer/console/preset/proc/get_preset_programs(var/app_preset_name)
+	for (var/datum/modular_computer_app_presets/prs in ntnet_global.available_software_presets)
+		//var/datum/modular_computer_app_presets/prs = new path()
+		if(prs.name == app_preset_name && prs.available == 1)
+			return prs.return_install_programs()
 
+// Override in child types to install preset-specific programs.
+/obj/machinery/modular_computer/console/preset/proc/install_programs(var/list/programs)
+	for (var/datum/computer_file/program/prog in programs)
+		cpu.hard_drive.store_file(prog)
 
 // ===== ADMIN CONSOLE =====
-/obj/machinery/modular_computer/console/preset/admin
+/obj/machinery/modular_computer/console/preset/all
 	console_department = "CC ITIQ"
-	desc = "A computer from the CC ITIQ Department. Only available on the odin"
+	desc = "A computer from the CC ITIQ Department. Only available on the Odin"
+	_app_preset_name = "all"
+	_enrolled = 1
 
 /obj/machinery/modular_computer/console/preset/admin/install_programs()
-	for(var/F in typesof(/datum/computer_file/program))
-		var/datum/computer_file/program/prog = new F
-		cpu.hard_drive.store_file(prog)
+	..()
 	cpu.battery_module = new/obj/item/weapon/computer_hardware/battery_module/lambda(cpu)
 
+// ===== Empty Test console CONSOLE =====
+/obj/machinery/modular_computer/console/preset/empty
+	 desc = "A stationary computer."
+
 // ===== ENGINEERING CONSOLE =====
 /obj/machinery/modular_computer/console/preset/engineering
 	 console_department = "Engineering"
 	 desc = "A stationary computer. This one comes preloaded with engineering programs."
-	 _software_locked = 1
-
-/obj/machinery/modular_computer/console/preset/engineering/install_programs()
-	cpu.hard_drive.store_file(new/datum/computer_file/program/power_monitor())
-	cpu.hard_drive.store_file(new/datum/computer_file/program/alarm_monitor())
-	cpu.hard_drive.store_file(new/datum/computer_file/program/atmos_control())
-	cpu.hard_drive.store_file(new/datum/computer_file/program/rcon_console())
-	cpu.hard_drive.store_file(new/datum/computer_file/program/chatclient())
-
+	 _app_preset_name = "engineering"
+	 _enrolled = 1
 
 // ===== MEDICAL CONSOLE =====
 /obj/machinery/modular_computer/console/preset/medical
 	 console_department = "Medbay"
 	 desc = "A stationary computer. This one comes preloaded with medical programs."
-	 _software_locked = 1
-
-/obj/machinery/modular_computer/console/preset/medical/install_programs()
-	cpu.hard_drive.store_file(new/datum/computer_file/program/chatclient())
-	cpu.hard_drive.store_file(new/datum/computer_file/program/suit_sensors())
-
+	 _app_preset_name = "medical"
+	 _enrolled = 1
 
 // ===== RESEARCH CONSOLE =====
 /obj/machinery/modular_computer/console/preset/research
```

Comment:
It would be a good idea to add subclasses to these presets, for the heads of staff in those departments.  Like /obj/machinery/modular_computer/console/preset/research/command
/obj/machinery/modular_computer/console/preset/medical/command

etc. For the apps that are limited to heads, like the exosuit thing